README.md

```diff
@@ -91,7 +99,12 @@ Write this grammar using EBNF with common extensions
 
 ## A2.2
 
-«replace this with your answer»
+<adjective> ::= "smelly" | "lazy"
+<noun> ::= "dog" | "cat"
+<verb> ::= "ate" | "ran"
+<adverb> ::= "slowly" | "noisily"
+
+sentence ::= "the" {<adjective>} <noun> <verb> [<adverb>]
 
 
 ## Q2.3 (6 points)
```

Comment:
The diagram FSM.pdf isn't valid: there's a line going from the transition into S1 to state S2.  In a state diagram, each movement between states is triggered by an event, and each state can only have one ongoing transition for a particular event (otherwise, how does it choose which one to take).

The diagram also doesn't allow for multiple adjectives.

You can fix it by removing state S2, and making S1 loop back to itself when it receives an adjective.

# -4